data/data/gcp/network/firewall-compute.tf

```diff
@@ -0,0 +1,154 @@
+  name    = "${var.cluster_id}-compute-ingress-vxlan"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "udp"
+    ports    = ["4789"]
+  }
+
+  source_tags = ["ocp-worker"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_vxlan_from_master" {
+  name    = "${var.cluster_id}-compute-ingress-vxlan-from-master"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "udp"
+    ports    = ["4789"]
+  }
+
+  source_tags = ["ocp-master"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_internal" {
+  name    = "${var.cluster_id}-compute-ingress-internal"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "tcp"
+    ports    = ["9000-9999"]
+  }
+
+  source_tags = ["ocp-worker"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_internal_from_master" {
+  name    = "${var.cluster_id}-compute-ingress-internal-from-master"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "tcp"
+    ports    = ["9000-9999"]
+  }
+
+  source_tags = ["ocp-master"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_internal_udp" {
+  name    = "${var.cluster_id}-compute-ingress-udp"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "udp"
+    ports    = ["9000-9999"]
+  }
+
+  source_tags = ["ocp-worker"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_internal_from_master_udp" {
+  name    = "${var.cluster_id}-compute-ingress-udp-from-master"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "udp"
+    ports    = ["9000-9999"]
+  }
+
+  source_tags = ["ocp-master"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_kubelet_insecure" {
+  name    = "${var.cluster_id}-compute-ingress-kubelet-insecure"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "tcp"
+    ports    = ["10250"]
+  }
+
+  source_tags = ["ocp-worker"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_kubelet_insecure_from_master" {
+  name    = "${var.cluster_id}-compute-ingress-kubelet-insecure-from-master"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "tcp"
+    ports    = ["10250"]
+  }
+
+  source_tags = ["ocp-master"]
+  target_tags = ["ocp-worker"]
+}
+
+resource "google_compute_firewall" "compute_ingress_services_tcp" {
+  name    = "${var.cluster_id}-compute-ingress-services-tcp"
+  network = google_compute_network.cluster_network.self_link
+
+  allow {
+    protocol = "tcp"
+    ports    = ["30000-32767"]
+  }
+
+  source_tags = ["ocp-worker"]
```

Comment:
nit: can we replace `ocp` with `os` as in `openshift` ?

Reply:
@abhinavdahiya How about ${var.cluster_id}-worker... will play better with multiple clusters in a vpc later on.

Reply:
Oh, yeah nice idea.